azure-ipam/logger/logger.go

```diff
@@ -1,61 +1,53 @@
 package logger
 
 import (
-	"strings"
-
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+const (
+	Filepath = "/var/log/azure-ipam.log"
 )
 
 type Config struct {
-	Level            string
-	OutputPaths      string // comma separated list of paths
-	ErrorOutputPaths string // comma separated list of paths
+	Level           string // Debug by default
+	Filepath        string // default /var/log/azure-ipam.log
+	MaxSizeInMB     int    // MegaBytes
+	MaxBackups      int    // # of backups, no limitation by default
 }
 
 // NewLogger creates and returns a zap logger and a clean up function
 func New(cfg *Config) (*zap.Logger, func(), error) {
-	loggerCfg := &zap.Config{}
-
-	level, err := zapcore.ParseLevel(cfg.Level)
+	logLevel, err := zapcore.ParseLevel(cfg.Level)
 	if err != nil {
 		return nil, nil, errors.Wrapf(err, "failed to parse log level")
 	}
-	loggerCfg.Level = zap.NewAtomicLevelAt(level)
-
-	loggerCfg.Encoding = "json"
-	loggerCfg.OutputPaths = getLogOutputPath(cfg.OutputPaths)
-	loggerCfg.ErrorOutputPaths = getErrOutputPath(cfg.ErrorOutputPaths)
-	loggerCfg.EncoderConfig = zapcore.EncoderConfig{
-		TimeKey:     "time",
-		MessageKey:  "msg",
-		LevelKey:    "level",
-		EncodeLevel: zapcore.LowercaseLevelEncoder,
-		EncodeTime:  zapcore.ISO8601TimeEncoder,
+	if cfg.Filepath == "" {
+		cfg.Filepath = Filepath
 	}
-
-	logger, err := loggerCfg.Build()
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "failed to build zap logger")
-	}
-
```

Comment:
why did we remove this custom encoder config?

Reply:
zap.conf.build() will return an error if the encoding or filepath have issues. But Zapcore.newcore() and zap.New(core) do not return error that I can capture. Do you have any specific scenario in mind that I will not capture an error?

Reply:
not sure i followed your comment. returning error is good right if there any issue instead of ignoring it?

Reply:
That customize encoder config was similar to  NewProductionEncoderConfig() which is already available in zap. you can edit the fields of the encoder returned by this function.  
like what I did here `encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder` 

And the error catch that has been removed is the result of build() which is no longer being used since we are using lumberjack and it seems we should use `Zapcore.newcore()` and `zap.New(core)` . These functions do not return error like build()

Reply:
got it.. we didn't set EncodeLevel: zapcore.LowercaseLevelEncoder in this new code...is that ok.?

Reply:
The constructor that I used is already setting that:
`func NewProductionEncoderConfig() zapcore.EncoderConfig {
	return zapcore.EncoderConfig{
		TimeKey:        "ts",
		LevelKey:       "level",
		NameKey:        "logger",
		CallerKey:      "caller",
		FunctionKey:    zapcore.OmitKey,
		MessageKey:     "msg",
		StacktraceKey:  "stacktrace",
		LineEnding:     zapcore.DefaultLineEnding,
		EncodeLevel:    zapcore.LowercaseLevelEncoder,
		EncodeTime:     zapcore.EpochTimeEncoder,
		EncodeDuration: zapcore.SecondsDurationEncoder,
		EncodeCaller:   zapcore.ShortCallerEncoder,
	}
}`

I also checked the log format and it seems fine. This is a sample:
`{"level":"debug","ts":"2022-08-24T03:33:38.371Z","msg":"Making request to CNS"}`